### PR TITLE
Use Cloudflare KV for storing & syncing pool data

### PR DESCRIPTION
**Pull request by dylandesrosier**

### CODE REVIEW

File: LICENSE_MIT [hunk 1]
```diff
@@ -22,4 +22,4 @@ SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
 CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
 OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR
 IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
-DEALINGS IN THE SOFTWARE.
+DEALINGS IN THE SOFTWARE.
```
Comment (chuckbergeron) [inline LICENSE_MIT:25]: 🤔 odd

File: packages/api-entrypoint/package.json [hunk 2]
```diff
@@ -1,43 +1,24 @@
 {
-  "name": "@pooltogether/api-entrypoint",
+  "private": true,
```
Comment (chuckbergeron) [inline packages/api-entrypoint/package.json:2]: Why `private: true` ?

File: packages/api-entrypoint/package.json [hunk 3]
```diff
@@ -1,43 +1,24 @@
...
+  "name": "api-yield-source-data",
```
Comment (chuckbergeron) [inline packages/api-entrypoint/package.json]: What was wrong with `@pooltogether/api-entrypoint` as a name? Shouldn't we also rename the `packages/api-entrypoint` folder if we're going to rename the project?

File: packages/api-runner/index.js [hunk 4]
```diff
@@ -1,31 +1,18 @@
-import { usePools, usePoolByAddress } from 'lib/hooks/usePool'
+import { usePools } from 'lib/hooks/usePool'
 import { usePoolContracts } from 'lib/hooks/usePoolContracts'
+import { usePoolByAddress } from 'lib/hooks/usePool'
```
Comment (chuckbergeron) [inline packages/api-runner/index.js]: That's odd. It's coming from the same file `lib/hooks/usePool` but they've been split into two separate imports 😸

File: packages/api-runner/lib/fetchers/getPoolChainData.js [hunk 5]
```diff
@@ -537,7 +556,7 @@ const formatPoolChainData = (
         pool.prize.externalErc721Awards,
         (erc721) => erc721.address === lootBoxAddress
       )
-      if (lootBoxes) {
+      if (lootBoxes && lootBoxes.length >= 1) {
```
Comment (chuckbergeron) [inline packages/api-runner/lib/fetchers/getPoolChainData.js]: Typically we use `lootBoxes.length > 0` everywhere. Does it need to be gte 1?

File: packages/api-runner/package.json [hunk 6]
```diff
@@ -58,12 +60,12 @@
     "@babel/plugin-transform-template-literals": "^7.13.0",
     "@ethersproject/abi": "^5.1.1",
     "@ethersproject/units": "^5.1.0",
-    "@pooltogether/current-pool-data": "3.4.11",
+    "@pooltogether/current-pool-data": "3.4.21",
```
Comment (chuckbergeron) [inline packages/api-runner/package.json:63]: Have `3.4.23` above. Should make this the same I imagine.

File: packages/api-yield-source-data/package.json [hunk 7]
```diff
@@ -0,0 +1,19 @@
+{
+  "private": true,
```
Comment (chuckbergeron) [inline packages/api-yield-source-data/package.json:2]: Another `private: true`

File: utils/getPoolsKey.js [hunk 8]
```diff
@@ -0,0 +1 @@
+export const getPoolsKey = (chainId) => `${chainId} - Pools`
```
Comment (chuckbergeron) [inline utils/getPoolsKey.js]: If this is a cache key should we not avoid using spaces in the string?

**Review by chuckbergeron — CHANGES_REQUESTED** [review]

**Comment by chuckbergeron** [comment]
So something I've noticed after just checking staging right now and comparing to production:

GUSD: Staging: GUSD $392
GUSD: Prod: $236   
DIFF: 39.79%

USDT: Staging: USDT $237
USDT: Prod: $198  
DIFF: 16.45%

The GUSD numbers are really different (40%), and USDT a bit different (16%). Maybe something snuck in the code or was removed that affects them?

**Comment by dylandesrosier** [comment]
The aave estimates are no longer hard coded. That could be the reason for the difference

**Comment by chuckbergeron** [comment]
Well that's a positive then 👍


--
Chuck Bergeron | PoolTogether
A new way to save
https://pooltogether.com

**Review by chuckbergeron — APPROVED** [review]

**Pull request merged**